### PR TITLE
Rework OIDC page intros.

### PR DESCRIPTION
**Pull request by gilesgas**
This PR was done to state up front the purpose of OIDC implementations in Buildkite Pipelines and Packages, rather than explaining what an OIDC token is first.

These are the amended pages:

- [OIDC in Buildkite Pipelines](https://buildkite.com/docs/pipelines/security/oidc) (preview [here](https://2925--bk-docs-preview.netlify.app/docs/pipelines/security/oidc))
- [OIDC in Buildkite Packages](https://buildkite.com/docs/packages/security/oidc) (preview [here](https://2925--bk-docs-preview.netlify.app/docs/packages/security/oidc))

### CODE REVIEW

**Comment by gilesgas** [comment]
Feedback from @mbelton-buildkite :

- Add initial intro paras on what OIDC is.

File: pages/platform/_oidc_introduction.md [hunk 1]
```diff
@@ -0,0 +1 @@
+[Open ID Connect (OIDC)](https://openid.net/developers/how-connect-works/) is an authentication protocol based on the [OAuth 2.0 framework](https://auth0.com/docs/authenticate/protocols/oauth/) that allows an _OIDC token_, containing metadata (or _claims_) about a user or object, which is issued by one system or service (as part of one or more actions), to be consumed and authenticated by another, such as a service delivered by a third-party or another offered by the same organization. An _OIDC policy_ configured on this other service defines which OIDC tokens (based on their claims) are permitted to perform the actions. If the OIDC token's claims match those of the OIDC policy configured in the other service, the token is authenticated and the service issuing the token is permitted to perform its actions on the other service.
```
Comment (tessereth) [inline pages/platform/_oidc_introduction.md:1]: This is accurate, though the opening sentence is a bit complex and hard to parse. Not that OIDC is ever easy to understand.

I do wonder if the word "signed" should stay in there somewhere. Anyone can make a _claim_, that doesn't really mean anything. A _signed_ claim is something you can do something with.

Here's my attempt to unravel the first sentence a bit. I don't know if it's better, but maybe it's useful food for thought.

```suggestion
[Open ID Connect (OIDC)](https://openid.net/developers/how-connect-works/) is an authentication protocol based on the [OAuth 2.0 framework](https://auth0.com/docs/authenticate/protocols/oauth/). With OIDC, one system or service issues an _OIDC token_ containing signed metadata (or _claims_) about a user or object. This token can be consumed by another service (which may be offered by a third-party or by the same organization) to authenticate the user or object. An _OIDC policy_ configured on this other service defines which OIDC tokens (based on their claims) are permitted to perform the actions. If the OIDC token's claims match those of the OIDC policy configured in the other service, the token is authenticated and the service issuing the token is permitted to perform its actions on the other service.
```

File: pages/pipelines/security/oidc/aws.md [hunk 2]
```diff
@@ -4,7 +4,7 @@ keywords: oidc, authentication, IAM, roles, AWS
 
 # OIDC with AWS
 
-The [Buildkite Agent's `oidc` command](/docs/agent/v3/cli-oidc) allows you to request an [Open ID Connect (OIDC)](https://openid.net/developers/how-connect-works/) token representing the current pipeline and its job. These tokens can be consumed by AWS and exchanged for an Identity and Access Management (IAM) role with AWS-scoped permissions.
+The [Buildkite Agent's `oidc` command](/docs/agent/v3/cli-oidc) allows you to request an [Open ID Connect (OIDC)](https://openid.net/developers/how-connect-works/) token containing _claims_ the current pipeline and its job. These tokens can be consumed by AWS and exchanged for an Identity and Access Management (IAM) role with AWS-scoped permissions.
```
Comment (tessereth) [inline pages/pipelines/security/oidc/aws.md:7]:
```suggestion
The [Buildkite Agent's `oidc` command](/docs/agent/v3/cli-oidc) allows you to request an [Open ID Connect (OIDC)](https://openid.net/developers/how-connect-works/) token containing _claims_ about the current pipeline and its job. These tokens can be consumed by AWS and exchanged for an Identity and Access Management (IAM) role with AWS-scoped permissions.
```

**Review by tessereth — APPROVED** [review]
Definitely looks better than before! Thanks for tidying all this up.

File: pages/platform/_oidc_introduction.md (hunk 1 above)
Comment (gilesgas) [reply pages/platform/_oidc_introduction.md:1]: Love this! Thanks @tessereth ! This reads a lot better 😄
